shell/platform/tizen/tizen_input_method_context.cc

```diff
@@ -191,6 +197,51 @@ bool TizenInputMethodContext::HandleEvasEventKeyUp(Evas_Event_Key_Up* event) {
       reinterpret_cast<Ecore_IMF_Event*>(&imf_event));
 }
 
+#ifdef NUI_SUPPORT
+bool TizenInputMethodContext::HandleNuiKeyEvent(const char* device_name,
+                                                uint32_t device_class,
+                                                uint32_t device_subclass,
+                                                const char* key,
+                                                const char* string,
+                                                uint32_t modifiers,
+                                                uint32_t scan_code,
+                                                size_t timestamp,
+                                                bool is_down) {
```

Comment:
This method (or `TizenViewNui::OnKey`) has too many parameters. They may be wrapped up into a class or struct but I'm not sure if it's a good solution.

Reply:
These are all Ecore_Events, but NUI doesn't provide all properties of EcoreEvent (maybe?).
I think we can add a `struct NuiEventKey`(?) structure, but I'd like to improve it in another PR if needed.